File: OOP/6_multilevel_inheritance.cpp
#include <bits/stdc++.h>
using namespace std;

// parent class
class A
{
public:
  string name;
  int age;

  // constructor
  A(string name, int age)
  {
    this->name = name;
    this->age = age;
  }

  // method
  void display()
  {
    cout << "Your is name" << name << " and age " << age << endl;
  }
};

// inherite from class A
class B:public A
{
public:
  string designation;
  B(string name, int age,string designation):A(name,age)
  {
    this->designation = designation;
  }
};

// inheriate from class B
class C:public B{
  public:
  string sub;
  C(string name, int age,string designation,string sub):B(name,age,designation){
    this->sub=sub;
  }
  void PersonC_info()
  {
    cout << "Your is name" << name << " and age " << age << " designation  "<<designation<< " Subject " <<sub<<endl;
  }

};

int main()
{

  C obj("John", 25, "student","Cse");
  obj.PersonC_info();
  return 0;
}

File: OOP/6_multilevel_inheritance.py
class A:
  def __init__(self,name,age) -> None:
    self.name=name
    self.age=age

  def display_A(self):
    print(f"{self.name} {self.age}")

# inherite from class A
class B(A):
  def __init__(self, name, age,designation) -> None:
    super().__init__(name, age)
    self.designation=designation
  
  def display_B(self):
    print(f"{self.name} {self.age} {self.designation}")

# inherite from clsss B
class C(B):
  def __init__(self, name, age, designation,sub) -> None:
    super().__init__(name, age, designation)
    self.sub=sub

  def display_C(self):
    print(f"{super().display_B()} {self.sub}")

person =C('rafiul',23,'student','cse')
person.display_C()


